scripts/dts/gen_defines.py

```diff
@@ -435,11 +435,23 @@ def irq_name_alias(irq, cell_name):
             alias += "_" + str2ident(cell_name)
         return alias
 
+    def encode_zephyr_multi_level_irq(irq, irq_num):
+        # See doc/reference/kernel/other/interrupts.rst for details
+        # on how this encoding works
+        irq_ctrl = irq.controller
+        while irq_ctrl.interrupts:
+            irq_num = (irq_num + 1) << 8
+            irq_num |= irq_ctrl.interrupts[0].specifier["irq"]
```

Comment:
Should check that the "irq" key appears in `irq_ctrl.interrupts[0].specifier`, and give a helpful error otherwise.

The goal is to never fail with a generic Python error in the new code.

Reply:
fixed.

Reply:
Could give the path to the binding (`irq_ctrl.binding_path`) instead/as well, since that's where it's missing.

```python
_err("expected 'irq' to appear in '#cells' in {} (for {!r})"
     .format(irq_ctrl.binding_path, irq_ctrl)) 
```

(Should've replied here instead.)